Generators refactor and code splitting

mmfallacy opened the pull request:
This PR aims to define general guidelines in refactoring `generators.py` and `solutions.py`. Changes have not yet been integrated with the main program.

# Changes
* Code-splitted `generators.py` into `generators\`
* Renamed `generators.py` into `_generators.py` and fixed imports
* **!!! Renamed `solutions` import into `_solutions` (please rename your `solutions.py`) !!!**
* Created generator and checker function templates
* Code splitted `auxi.py` into `utils\`

mmfallacy committed:
CHORE: code split generators; FEAT: TCGenerator object that creates random testcase generators given a [TYPE][NUM] problem string
# See \generators\__init__.py for more information

mmfallacy committed:
CHORE: Create _genTemplate for future refactoring of generators.py

mmfallacy committed:
DOCS: Add documentation for TCGenerator class

mmfallacy committed:
FEAT: Create solutions directory to hold checker functions; _solutionsTemplate for future refactoring;

CHORE: Create .gitignore to ignore mock files, pycache and .pyc files

mmfallacy committed:
CHORE: Rename generators.py to _generators.py so generators\ will not yet be used. Fixed imports for generators and _solutions

mmfallacy committed:
CHORE: Code split auxi.py into utils\

mmfallacy committed:
CHORE: Untrack generators\__pycache__

mmfallacy committed:
Merge branch 'refactor/code-splitting' of https://github.com/smtnhacker/OJHacker into refactor/code-splitting

BastiDood reviewed (changes requested):
Super concerned lang ako sa fact na maraming duplicate file names and stuff. It doesn't look _right_ na may mga "underscored" versions yung ilang modules. It's probably best to rename them outright.

smtnhacker committed:
Integrated TC Generator to main.py and few updates

Rewrote main.py to use the TC Generator, instead of the mock _generator.py. Also, transferred the *Template.py to their respected readme.md files. Also added a couple of PA generators and added dummy solutions (will get updated in replit in for the time being).

smtnhacker committed:
Added generators

Added Generator for LE09, LE12, and PA08. Also added another list of constants for a generator titled locations.py.

smtnhacker committed:
UPDATE: Added generators

Added generators for LE10, LE11, and PA07. Note that the parameter for the checker in LE10 is different.

BastiDood committed:
Chore: delete `_generators.py`

BastiDood committed:
Chore: rename README file names for consistency

BastiDood committed:
Merge: pull in changes from main and resolve conflicts

BastiDood committed:
Chore: move auxiliary functions to utilities

BastiDood committed:
Chore: remove file name comment

BastiDood committed:
Refactor: integrate generator refactor

BastiDood committed:
Refactor: use single underscores to indicate field privacy

BastiDood committed:
Format: use underscores to indicate unused variables

smtnhacker committed:
Refactor: Move commads to cogs

smtnhacker committed:
Fix: Correct argument parsing

smtnhacker committed:
Chore: Minor corrections
